qiita_db/support_files/patches/33.sql

```diff
@@ -0,0 +1,435 @@
+-- September 4, 2015
+-- Change the database structure to remove the RawData, PreprocessedData and
+-- ProcessedData division to unify it into the Artifact object
+
+-- Creating the new tables
+
+-- Rename the table filetype
+ALTER TABLE qiita.filetype RENAME TO artifact_type;
+ALTER TABLE qiita.artifact_type RENAME COLUMN filetype_id TO artifact_type_id;
+ALTER TABLE qiita.artifact_type RENAME COLUMN type TO artifact_type;
+ALTER TABLE qiita.artifact_type ADD description varchar;
+
+-- Rename the processed_data_status table
+ALTER TABLE qiita.processed_data_status RENAME TO visibility;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status_id TO visibility_id;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status TO visibility;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status_description TO visibility_description;
+UPDATE qiita.visibility
+    SET visibility_description = 'Only visible to the owner and shared users'
+    WHERE visibility = 'private';
+UPDATE qiita.visibility
+    SET visibility_description = 'Visible to everybody'
+    WHERE visibility = 'public';
+
+-- Artifact table - holds an abstract data object from the system
+CREATE TABLE qiita.artifact (
+    artifact_id                         bigserial  NOT NULL,
+    generated_timestamp                 timestamp  NOT NULL,
+    command_id                          bigint  ,
+    command_parameters_id               bigint  ,
+    visibility_id                       bigint  NOT NULL,
+    file_status                         varchar  NOT NULL,
```

Comment:
That is because if an error occurs, it is stored in this variable. That is what we've been doing so far, and we still have not found a good alternative for this.